### ISSUE TITLE
EXCEPTION: St9bad_alloc std::bad_alloc C:\Users\Papa\Desktop\MintCoin-Qt.exe in Runaway exception

### ISSUE THREAD

**Issue by shane-kerr** [MintcoinCommunity/Mintcoin-Desktop-Wallet#67]
Many Windows users have reported:

```
EXCEPTION: St9bad_alloc
std::bad_alloc
C:\Users\Papa\Desktop\MintCoin-Qt.exe in Runaway exception
```

A workaround for now is: 

- Make a shortcut to Mintcoin-Qt.exe
- Right click on the _shortcut->Properties_. Then add `-splash=0` to the end of the target string.

We should find and fix the underlying issue though.

**Comment by gradegal** [comment]
thanks for the workaround - now this exception doesn`t show up again - but the wallet gets no connections to network.


[debug.log](https://github.com/MintcoinCommunity/Mintcoin-Desktop-Wallet/files/2050054/debug.log)

**Comment by shane-kerr** [comment]
The code in the GitHub repository has a fix for not getting connections, but for now you need to specify nodes directly using `addnode`. You may find one of these videos helpful:

(Windows) https://www.youtube.com/watch?v=bc7FecSGm1s
(Mac) https://www.youtube.com/watch?v=yn4M_H69ZP8
(Mac Lion to High Sierra) https://www.youtube.com/watch?v=MQY1KOyu8uY

**Comment by gradegal** [comment]
....but for now you need to specify nodes directly using addnode.

How can I do this ?

**Comment by Myke500** [comment]
Getting a slightly different error
EXCEPTION: St13runtime_error
CDE(): error opening database environment IO error: 

Same or needs new issue thread?

**Comment by shane-kerr** [comment]
It seems that we were running up into the default 2 GB process size limit for 32-bit applications in Windows. 32-bit applications running in 32-bit Windows can support up to 3 GB of RAM, and 32-bit applications running in 64-bit Windows can support up too 4 GB of RAM, but they need to be compiled with this feature enabled.

Here's the Microsoft page about this build option, IMAGE_FILE_LARGE_ADDRESS_AWARE, for your reading pleasure:

https://docs.microsoft.com/en-gb/windows/desktop/Memory/memory-limits-for-windows-releases#memory_limits

In our case we are using the MXE cross-compile from Linux, and need to add the `-Wl,--large-address-aware` flag to our 32-bit Windows build.

**Comment by shane-kerr** [comment]
This seems to be fixed with binaries built by Travis, which we will use for the next release.

I'll keep this ticket open to point people to until then.

**Comment by gradegal** [comment]
Dear Shane,
thank you very much for this update 👍🏼
I am eager for the next release of the wallet.

**Comment by shane-kerr** [comment]
This is fixed with the release of the 2.1 wallet.
